apps/website/src/hooks/consent.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState,
+  createContext,
+  useContext,
+  useMemo,
+} from "react";
+
+const defaultConsent = {
+  twitch: false,
+  youtube: false,
+} as const;
+
+type ConsentKeys = keyof typeof defaultConsent;
+
+export type Consent = Readonly<{
+  [K in ConsentKeys]: boolean;
+}>;
+
+const parseConsent = (
+  obj: Record<string, unknown>,
+  existing?: Consent
+): Consent =>
+  Object.freeze(
+    (Object.keys(existing || defaultConsent) as ConsentKeys[]).reduce(
+      (acc, key) => ({
+        ...acc,
+        [key]:
+          typeof obj[key] === "boolean"
+            ? obj[key]
+            : (existing || defaultConsent)[key],
+      }),
+      {}
+    )
+  ) as Consent;
+
+export type ConsentContext = Readonly<{
+  consent: Consent;
+  updateConsent: (consent: Partial<Consent>) => void;
+  consentLoaded: boolean;
```

Comment:
I never know with hooks whether the pattern should be that properties have names that indicate where they are from (as they are currently), or whether they should have clean names (e.g. `status, update, loaded` here) -- thoughts?

Reply:
Went with clean names, it was getting messy with them all having the word consent in it